### PR TITLE
REF: Wallet Quick Actions code isolation

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Review by Overtorment — APPROVED** [review]

**Comment by Overtorment** [comment]
linter & unit failed tho.
plus would not hurt to convert  to ts

**Comment by marcosrdz** [comment]
> linter & unit failed tho. plus would not hurt to convert to ts

Converted to TSX

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ixiifiiqnhryrrzgawwthoxdhy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11863)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ei3ie7l3omfrsmy7qsj2wohgju?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11864)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/32fopd6mtpyctlvupkoac5oygq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11865)

**Review by Overtorment — CHANGES_REQUESTED** [review]
E2E failed

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: class/quick-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    if (walletsInitialized) {
+      DeviceEventEmitter.addListener('quickActionShortcut', walletQuickActions);
+      popInitialShortcutAction().then(popInitialAction);
+      return () => DeviceEventEmitter.removeAllListeners('quickActionShortcut');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [walletsInitialized]);
+
+  // @ts-ignore: Fix later
+  DeviceQuickActions.setEnabled = async (enabled: boolean = true): Promise<void> => {
```
Comment (Overtorment) [inline class/quick-actions.tsx:44]: I think methods should be outside func body

**Review by Overtorment — APPROVED** [review]

File: class/quick-actions.tsx (hunk 1 above)
Comment (marcosrdz) [reply class/quick-actions.tsx:44]: Ill refactor those so they are modern and compatible with intellisense

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**